compiler/rustc_resolve/src/ident.rs

```diff
@@ -812,6 +812,24 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             ignore_import,
         )
     }
+
+    fn resolve_super_in_module(
+        &self,
+        ident: Ident,
+        mut module: Module<'ra>,
+        determinacy: Determinacy,
```

Comment:
The logic looks correct now, but `Determinacy` is something that is entirely unrelated to this case.

You could pass `Option<Module>, &ParentScope`, or even `Result<Module, &ParentScope>` instead.